examples/chart/teleport-kube-agent/values.yaml

```diff
@@ -57,6 +57,17 @@ podSecurityPolicy:
 # Labels is a map of key values pairs about this cluster
 labels: {}
 
+################################################################
+# Values that must be provided if using persistent storage for Teleport state
+# enabled: Whether to use persistent storage
+# storageClassName: The name of the kubernetes storage class to use when creating volumes. See https://kubernetes.io/docs/concepts/storage/storage-classes/
+# requests: The size of the volume to request from the persistent storage system
+################################################################
+storage:
+  enabled: false
+  storageClassName: ""
+  requests: 128Mi
```

Comment:
I actually have no idea how big the teleport storage needs are.
Do you have any numbers on the real-world size?
If this needs to be increased after deployment, will existing data be preserved?

Reply:
> I actually have no idea how big the teleport storage needs are.
> Do you have any numbers on the real-world size?
Last 7 days I've had this deployed / idle it's used 19MiB. I may have assumed incorrectly that because this is a node agent and logs/audit/etc are meant to be centralized that the default for this volume can be relatively small.

> If this needs to be increased after deployment, will existing data be preserved?
It depends on how the storage class is setup and presumably whether the block driver supports it. The storage class has a boolean field called allowVolumeExpansion, and if it's set, the volume can be expanded without data loss. It will interrupt the pod temporarily since the volume needs to be unmapped and the filesystem expanded.

Reply:
I think this volume can be fairly small - no logs are really kept locally (they all go to Docker/containerd) and the sqlite database for the agent itself should only be holding a keypair and some cache data.